product/specs/2022-01-views/02-modeling-views.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Views: Modeling Views
+description:
+published: true
+date: 2022-01-13T19:49:54Z
+tags: 
+editor: markdown
+dateCreated: 2022-01-13T19:49:54Z
+---
+
+Here's how I think we should model views in our API and UI. Each heading represents an attribute of Views.
+
+Before I get into attributes, here's an example table named `Movies` that I'll use to illustrate different types of Views.
+
+| ID | Title | Year | Genre |
+|-|-|-|-|
+| 1 | The River Wild | 1994 | Thriller |
+| 2 | Don't Look Up | 2021 | Comedy |
+| 3 | Daylight | 1996 | Action |
+| 4 | Jason Bourne | 2016 | Action |
+
+## Query
+This is the SQL query that defines the view. This is **required**.
+
+## Columns
+Output columns of the View's Query, they will be shown in the UI.  At least one column is **required** for a view.
+
+Column attributes are defined [in a separate page](/product/specs/2022-01-views/03-modeling-view-columns.md).
+
+## Rows
+These are the output rows of the View's Query. Rows are **not required**, Views can contain 0 rows.
+
+## Filters
+Views can have filters applied. Unlike Tables, view filters are not necessarily related to the columns that are present in the view.
+
+Using the example table above, imagine a view created from the query `SELECT ID, Title FROM Movies WHERE Year > 2000;` This will return this view: which is filtered by Year even though it's not a column in the View.
+
+| ID | Title |
+|-|-|
+| 2 | Don't Look Up |
+| 4 | Jason Bourne |
+
+## Sorting
+Views can have sorting applied. Unlike Tables, view sorting is not necessarily related to the columns that are present in the view.
+
+Using the example table above, imagine a view created from the query `SELECT ID, Title FROM Movies ORDER BY Year;` This will return this view, which is ordered by Year even though it's not a column in the View.
```

Comment:
Based on the same reasoning as the filters, I do not think we should be showing this sorting in the 'sort dropdown' on the frontend.

If the user sorts the view, I would assume the backend would sort it in addition to the existing sorting for Year. In essence, if they sort by Title, the sorting would be prioritized first by Title and then by Year (which would function as kind of a default sort).

As mentioned in the filters comments, if we do not intend to show the sorting applied in the query on the UI, I think it's best to explicitly mention that here.

Reply:
We can discuss this in one place under filtering, I think whatever conclusion we get to will apply.